### PR TITLE
Clean-up constraint generation for insert builtin function

### PR DESCRIPTION
**Pull request by romainbrenguier**
No functional changes.
This is a follow-up for https://github.com/diffblue/cbmc/pull/5035. Some code is now unreachable and can be removed, other parts can be moved to the string_insertion_builtin_function class to improve the structure of the code.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/solvers/strings/string_insertion_builtin_function.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  /// inserted \c input at position \c offset given by the first argument.
+  /// We write offset' for `max(0, min(res.length, offset))`.
+  /// These axioms are:
+  /// 1. res.length = s1.length + s2.length
+  /// 2. forall i < offset' . res[i] = s1[i]
+  /// 3. forall i < s2.length. res[i + offset'] = s2[i]
+  /// 4. forall i in [offset', s1.length). res[i + s2.length] = s1[i]
+  /// This is equivalent to
+  /// `res=concat(substring(s1, 0, offset'), concat(s2, substring(s1, offset')))`.
+  /// \param fresh_symbol: generator of fresh symbols
```
Comment (peterschrammel) [inline src/solvers/strings/string_insertion_builtin_function.h]: 🚫 docs don't match signature

File: src/solvers/strings/string_insertion_builtin_function.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  ///   `result.length = input1.length + input2.length`
+  exprt length_constraint() const override;
+
+  bool maybe_testing_function() const override
+  {
+    return false;
+  }
+
+protected:
+  explicit string_insertion_builtin_functiont(
```
Comment (peterschrammel) [inline src/solvers/strings/string_insertion_builtin_function.h]: explicit not required

**Review by peterschrammel — APPROVED** [review]

**Review by smowton — APPROVED** [review]

**Review by martin-cs — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: fc4baa4b50d7232616286bfe6a13875c4a716d27).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124918171

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 3e1096d385f7578b59bb061feb86fb8107b98820).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124929261

**Pull request merged**